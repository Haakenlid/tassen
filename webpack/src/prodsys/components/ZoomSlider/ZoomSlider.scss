.ZoomSlider {
  display: flex;
  flex-flow: row wrap;
  .iconwrapper {
    padding: 0.4rem;
    display: flex;
    flex-direction: column;
    align-items: center;
    small {
      margin: 0.3em 0;
      font-size: calc(0.8rem + 0.05em);
    }
  }
  .icon {
    margin: 0 0.1em;
    height: 1em;
    width: 1em;
  }
}
