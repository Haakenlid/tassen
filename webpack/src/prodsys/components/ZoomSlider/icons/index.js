export Microscope from './Microscope.js'
export Watch from './Watch.js'
export Mobile from './Mobile.js'
export Preview from './Preview.js'
export Laptop from './Laptop.js'
export Desktop from './Desktop.js'
export Telescope from './Telescope.js'
