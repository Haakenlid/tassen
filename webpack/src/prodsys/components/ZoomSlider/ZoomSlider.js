// import * as Icons from './icons/'
import { Binoculars, Mobile, Eye, Desktop, Laptop } from 'components/Icons'

const ZoomSlider = props => (
  <div className="ZoomSlider" {...props}>
    {R.values(
      R.mapObjIndexed(
        (Icon, name) => (
          <div className="iconwrapper" key={name}>
            <Icon className="icon" />
            <small>{name}</small>
          </div>
        ),
        { Mobile, Eye, Laptop, Desktop, Binoculars },
      ),
    )}
  </div>
)

export default ZoomSlider
