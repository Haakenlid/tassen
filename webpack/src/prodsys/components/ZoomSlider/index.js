import ZoomSlider from './ZoomSlider.js'
import './ZoomSlider.scss'
export default ZoomSlider
