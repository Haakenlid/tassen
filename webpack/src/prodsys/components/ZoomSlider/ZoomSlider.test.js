import ZoomSlider from './index'
import renderer from 'react-test-renderer'

describe('ZoomSlider', () => {
  test('renders correctly', () => {
    const tree = renderer.create(<ZoomSlider />).toJSON()
    expect(tree).toEqual(expect.any(Object))
    expect(tree).toMatchSnapshot()
  })
})
