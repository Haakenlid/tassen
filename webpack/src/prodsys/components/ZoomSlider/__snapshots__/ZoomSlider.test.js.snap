// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ZoomSlider renders correctly 1`] = `
<div
  className="ZoomSlider"
>
  <div
    className="iconwrapper"
  >
    <svg
      className="icon"
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 24 24"
      width="1em"
    >
      <path
        d="M16 1H8C6.34 1 5 2.34 5 4v16c0 1.66 1.34 3 3 3h8c1.66 0 3-1.34 3-3V4c0-1.66-1.34-3-3-3zm-2 20h-4v-1h4v1zm3.25-3H6.75V4h10.5v14z"
      />
    </svg>
    <small>
      Mobile
    </small>
  </div>
  <div
    className="iconwrapper"
  >
    <svg
      className="icon"
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 24 24"
      width="1em"
    >
      <path
        d="M12 4.5C7 4.5 2.73 7.61 1 12c1.73 4.39 6 7.5 11 7.5s9.27-3.11 11-7.5c-1.73-4.39-6-7.5-11-7.5zM12 17c-2.76 0-5-2.24-5-5s2.24-5 5-5 5 2.24 5 5-2.24 5-5 5zm0-8c-1.66 0-3 1.34-3 3s1.34 3 3 3 3-1.34 3-3-1.34-3-3-3z"
      />
    </svg>
    <small>
      Eye
    </small>
  </div>
  <div
    className="iconwrapper"
  >
    <svg
      className="icon"
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 24 24"
      width="1em"
    >
      <path
        d="M22 18V3H2v15H0v2h24v-2h-2zm-8 0h-4v-1h4v1zm6-3H4V5h16v10z"
      />
    </svg>
    <small>
      Laptop
    </small>
  </div>
  <div
    className="iconwrapper"
  >
    <svg
      className="icon"
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 24 24"
      width="1em"
    >
      <path
        d="M21 2H3c-1.1 0-2 .9-2 2v12c0 1.1.9 2 2 2h7v2H8v2h8v-2h-2v-2h7c1.1 0 2-.9 2-2V4c0-1.1-.9-2-2-2zm0 14H3V4h18v12z"
      />
    </svg>
    <small>
      Desktop
    </small>
  </div>
  <div
    className="iconwrapper"
  >
    <svg
      className="icon"
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 512 512"
      width="1em"
    >
      <path
        d="M416 48c0-8.84-7.16-16-16-16h-64c-8.84 0-16 7.16-16 16v48h96V48zM63.91 159.99C61.4 253.84 3.46 274.22 0 404v44c0 17.67 14.33 32 32 32h96c17.67 0 32-14.33 32-32V288h32V128H95.84c-17.63 0-31.45 14.37-31.93 31.99zm384.18 0c-.48-17.62-14.3-31.99-31.93-31.99H320v160h32v160c0 17.67 14.33 32 32 32h96c17.67 0 32-14.33 32-32v-44c-3.46-129.78-61.4-150.16-63.91-244.01zM176 32h-64c-8.84 0-16 7.16-16 16v48h96V48c0-8.84-7.16-16-16-16zm48 256h64V128h-64v160z"
      />
    </svg>
    <small>
      Binoculars
    </small>
  </div>
</div>
`;
